### ISSUE TITLE
Check if `with pytest.raises()` CM block has exactly one statement inside

### ISSUE THREAD

**Issue by webknjaz** [wemake-services/wemake-python-styleguide#1110]
# Rule request

I want a rule that prevents users from writing potentially incorrect tests.

## Thesis

This is what I have in mind.

Correct use:
```python
def test_xxx():
    tested_thing = do_some_init()
    with pytest.raises(SomeError):
        tested_thing.some_meth()  # Should raise SomeError
```

Potentially problematic use:
```python
def test_xxx():
    with pytest.raises(SomeError):
        tested_thing = do_some_init()  # <- even if the test worked properly when first comitted, there's a chance that this line will start raising SomeError at some point in the future making the next one unreachable, thus turning this test into an incorrect one
        tested_thing.some_meth()  # Should raise SomeError
```

## Reasoning

This would prevent silly mistakes in tests.

P.S. I'm not sure if this is the right place to request such linter but I wanted to document this idea somewhere because it comes up in code reviews quite often and many people don't really realize that it may cause problems.

**Comment by sobolevn** [comment]
Thanks, @webknjaz! This is a good idea.

The same also applies to `with suppress():` from `functools`. This will be covered with `typed-linter` soon. Because currently we don't like the idea to base on names, not types.

We try not to include any package specific checks inside this project. It is better to support this feature in https://github.com/m-burst/flake8-pytest-style

I have opened a new issue there: https://github.com/m-burst/flake8-pytest-style/issues/1
